### PR TITLE
add support for pytest and use it on travis/appveyor

### PR DESCRIPTION
**Pull request by pkienzle**
The existing repo only checks the models, and doesn't run the rest of the tests in the repo.

This patch adds support for pytest (since nose is no longer being developed, and nose2 didn't work) to the travis-ci and appveyor builds.  

Now tests such as the mixture model tests added to the core by @rprospero in #61 will be run during the build, and commits such as 8698a0ddea5 by @pkienzle (which broke mixtures of oriented shapes) won't go undetected so long.

### CODE REVIEW

**Comment by pkienzle** [comment]
It should be possible to write a test discovery mechanism as part of ```conftest.py``` that converts yield tests into a normal list of tests so that there is a path forward when pytest 4 removes the feature.  

I think coding ```pytest_pycollect_makeitem``` will allow it but it is not well documented how to write it.

**Comment by pkienzle** [comment]
Okay, pytest<4 restriction removed by moving the generator collection within conftest.

**Comment by wpotrzebowski** [comment]
Test button from OpenCL options doesn't work with the pytest branch. It seems to be due to the change in kernelcl interface. Error: global name 'cl' is not defined

**Comment by wpotrzebowski** [comment]
Two tests: model_tests[104]::raspberry-opencl and model_tests[106]::rectangular_prism-opencl are failing on my machine, even though they pass "old" model testing. It may be something with my set-up though.

File: conftest.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            if description:
+                # Strip spaces from start and end and strip dots from end
+                # pytest converts '.' to '::' on output for some reason.
+                description = description.strip().rstrip('.')
+                # Join multi-line descriptions into a single line
+                if '\n' in description:
+                    description = " ".join(line.strip() for line in description.split('\n'))
+
+            # Set the description as part of the node identifier.
+            if description:
```
Comment (wpotrzebowski) [inline conftest.py]: I think it can be in the same if statement as above

**Comment by pkienzle** [comment]
Test button for OpenCL should now work.  

Some caveats: 

* clicking test will show an error when no OpenCL is selected
* turning on/off OpenCL does not reset the current model; the Compute button will show an error until a new model is selected.
* no longer need reload(sasmodels.core) or kernelcl.ENV = None in sasview; instead use kernelcl.reset_environment()

I'm not going to try to fix these.

**Pull request merged**